File: trackflow-backend/src/main/java/com/trackflow/module/user/service/UserService.java
package com.trackflow.module.user.service;

import com.trackflow.module.user.dto.CreateUserRequest;
import com.trackflow.module.user.dto.UserResponse;
import com.trackflow.module.user.entity.UserRole;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface UserService {

    UserResponse createUser(CreateUserRequest request);

    UserResponse getUserById(UUID id);

    Page<UserResponse> getAllUsers(Pageable pageable);

    void updateUserRole(UUID id, UserRole role);

    void updateUserStatus(UUID id, boolean isActive);
}

File: trackflow-frontend/src/services/userService.js
import api from './api'

export const userService = {
  getAllUsers: (params) => api.get('/users', { params }),
  createUser: (data) => api.post('/users', data),
  getUserById: (id) => api.get(`/users/${id}`),
  updateRole: (id, role) => api.patch(`/users/${id}/role`, { role }),
  updateStatus: (id, isActive) => api.patch(`/users/${id}/status`, { isActive }),
  resetPassword: (id, password) => api.patch(`/users/${id}/password`, { password }),
}